msal_extensions/libsecret.py

```diff
@@ -0,0 +1,64 @@
+"""Implements a Linux specific TokenCache, and provides auxiliary helper types.
+
+This module depends on pygbject. But `pip install pygobject` would typically fail,
+until you install its dependencies first. For example, on a Debian Linux, you need::
+
+    sudo apt install libgirepository1.0-dev libcairo2-dev python3-dev gir1.2-secret-1
+    pip install pygobject
+
+Alternatively, you could skip Cairo & PyCairo, but you still need to do all these
+(derived from https://gitlab.gnome.org/GNOME/pygobject/-/issues/395)::
+
+    sudo apt install libgirepository1.0-dev python3-dev gir1.2-secret-1
+    pip install wheel
+    PYGOBJECT_WITHOUT_PYCAIRO=1 pip install --no-build-isolation pygobject
+"""
+import gi  # https://pygobject.readthedocs.io/en/latest/getting_started.html
+
+gi.require_version("Secret", "1")  # Would require a package gir1.2-secret-1
+# pylint: disable=wrong-import-position
+from gi.repository import Secret  # Would require a package gir1.2-secret-1
+
+class LibSecretAgent(object):
+    """A loader/saver built on top of low-level libsecret"""
+    # Inspired by https://developer.gnome.org/libsecret/unstable/py-examples.html
+    def __init__(  # pylint: disable=too-many-arguments
+            self,
+            schema_name,
+            attributes,  # {"name": "value", ...}
+            label="",  # Helpful when visualizing secrets by other viewers
+            attribute_types=None,  # {name: SchemaAttributeType, ...}
+            collection=None,  # None means default collection
+            ):  # pylint: disable=bad-continuation
+        self._collection = collection
+        self._attributes = attributes
+        self._label = label
+        self._schema = Secret.Schema.new(schema_name, Secret.SchemaFlags.NONE, {
+            k: (attribute_types or {}).get(k, Secret.SchemaAttributeType.STRING)
+            for k in attributes})
+
+    def save(self, data):
+        """Store data. Returns a boolean of whether operation was successful."""
+        return Secret.password_store_sync(
+            self._schema, self._attributes, self._collection, self._label,
+            data, None)
+
+    def load(self):
+        """Load a password in the secret service, return None when found nothing"""
+        return Secret.password_lookup_sync(self._schema, self._attributes, None)
+
+    def clear(self):
+        """Returns a boolean of whether any passwords were removed"""
+        return Secret.password_clear_sync(self._schema, self._attributes, None)
+
+def verify():
+    """Exception will be raised if libsecret is not functioning"""
+    agent = LibSecretAgent("Schema_Name Testing", {"attr1": "foo", "attr2": "bar"})
```

Comment:
Nit: use some constants like "persistence_test"

Reply:
IMHO, unless a string would be subjected to future localization need (which it isn't in our SDK scenarios), the pattern of "use pre-defined constants rather than repeated on-the-spot magic strings" does not give a real benefit, when that magic string is used once and only once.